packages/core/src/browser/tree/tree-widget.ts

```diff
@@ -470,11 +476,25 @@ export class TreeWidget extends VirtualWidget implements StatefulWidget {
 
     protected handleClickEvent(node: ITreeNode | undefined, event: MouseEvent): void {
         if (node) {
-            if (ISelectableTreeNode.is(node)) {
-                this.model.selectNode(node);
-            }
-            if (this.isExpandable(node)) {
-                this.model.toggleNodeExpansion(node);
+            if (!!this.props.multiSelect) {
+                const multi = this.isMulti(event);
+                if (ISelectableTreeNode.is(node)) {
```

Comment:
looks like there should be something like `toggleSelection(node)` on the tree model